tests/gtest/test_cli_daemon.cpp

```diff
@@ -160,4 +163,89 @@ TEST(CliDaemon, InputCommandLineCorrectly_AfterReveivingOnSessionSocket)
     cliDaemon.Deinit();
 }
 
+class CliDaemonTestOutput : public otbr::CliDaemon::Dependencies
+{
+public:
+    // Store a pointer to the CliDaemon to call HandleCommandOutput
+    CliDaemon *mCliDaemonInstance = nullptr;
+
+    otbrError InputCommandLine(const char *aLine) override
+    {
+        OTBR_UNUSED_VARIABLE(aLine);
+
+        if (mCliDaemonInstance != nullptr)
+        {
+            mCliDaemonInstance->HandleCommandOutput(kTestOutput);
+        }
+        return OTBR_ERROR_NONE;
+    }
+};
+
+TEST(CliDaemon, HandleCommandOutputCorrectly_AfterReveivingOnSessionSocket)
+{
+    const char *command    = "test command";
+    const char *netIfName  = "tun0";
+    const char *socketFile = "/run/openthread-tun0.sock";
+
+    CliDaemonTestOutput cliDependency;
+    CliDaemon           cliDaemon(cliDependency);
+    cliDependency.mCliDaemonInstance = &cliDaemon;
+
+    otbr::MainloopContext context;
+
+    EXPECT_EQ(cliDaemon.Init(netIfName), OT_ERROR_NONE);
+
+    int clientSocket = -1;
+    {
+        struct sockaddr_un serverAddr;
+
+        clientSocket = socket(AF_UNIX, SOCK_STREAM | SOCK_NONBLOCK, 0);
+        ASSERT_GE(clientSocket, 0) << "socket creation failed: " << strerror(errno);
+
+        memset(&serverAddr, 0, sizeof(serverAddr));
+        serverAddr.sun_family = AF_UNIX;
+        strncpy(serverAddr.sun_path, socketFile, sizeof(serverAddr.sun_path) - 1);
+        ASSERT_EQ(connect(clientSocket, (struct sockaddr *)&serverAddr, sizeof(serverAddr)), 0);
+
+        int rval = send(clientSocket, command, strlen(command), 0);
+        ASSERT_GE(rval, 0) << "Error sending command: " << strerror(errno);
+    }
+
+    char recvBuf[kCliMaxLineLength];
+    bool outputReceived = false;
+
+    while (!outputReceived)
+    {
+        context.mMaxFd   = -1;
+        context.mTimeout = {100, 0};
+        FD_ZERO(&context.mReadFdSet);
+        FD_ZERO(&context.mWriteFdSet);
+        FD_ZERO(&context.mErrorFdSet);
+
+        cliDaemon.UpdateFdSet(context);
+
+        FD_SET(clientSocket, &context.mReadFdSet);
+        context.mMaxFd = std::max(context.mMaxFd, clientSocket);
```

Comment:
We can use `context.AddToReadSet(clientSocket)`. 

https://github.com/openthread/ot-br-posix/blob/main/src/common/mainloop.hpp#L58